demos/frontend-react/app/src/components/ResultsViewer.js

```diff
@@ -0,0 +1,140 @@
+
+  convertOutputToTable(filePath) {
+    socket.emit('readCsvContent', filePath);
+    socket.on('readCsvContentRes', (items) => {
+      console.log('readCsvContentRes', items);
+      var col = [];
+      for (let i = 0; i < items.length; i++) {
+        for (var key in items[i]) {
+          if (col.indexOf(key) === -1) {
+            col.push(key);
+          }
+        }
+      }
+      var table = document.createElement("table");
+      var tr = table.insertRow(-1);
+      for (let i = 0; i < col.length; i++) {
+        var th = document.createElement("th");
+        th.setAttribute("style", "text-align:center;");
+        th.innerHTML = col[i];
+        tr.appendChild(th);
+      }
+      for (let i = 0; i < items.length; i++) {
+        tr = table.insertRow(-1);
+        for (var j = 0; j < col.length; j++) {
+          var tabCell = tr.insertCell(-1);
+          tabCell.innerHTML = items[i][col[j]];
+        }
+      }
+      document.getElementById("display").innerHTML = "";
+      document.getElementById("display").appendChild(table);
+    });
+  }
+
+  convertOutputToPureHtml(filePath) {
+    socket.emit('readUrlContent', filePath);
+    socket.on('readUrlContentRes', (val) => {
+      document.getElementById("display").innerHTML = val;
+    });
+  }
+
+  convertOutputToHtml(filePath) {
+    let htmlCode = "";
+
+    if (!filePath)
+      return htmlCode;
+
+    let ext = filePath.split('.').pop();
+    switch (ext) {
+      case 'png':
+        htmlCode = this.convertOutputToImage(filePath);
+        break;
+      case 'csv':
+        htmlCode = this.convertOutputToTable(filePath);
+        break;
+      case 'html':
+        htmlCode = this.convertOutputToPureHtml(filePath);
+        break;
+      default:
+        htmlCode = "";
+        break;
+    }
+    return htmlCode;
+  }
+
+  convertPathToPublic(pathToConvert) {
+    if (pathToConvert === null)
+      return pathToConvert;
+
+    // Hack: make all local path pulic
+    var localDir = '//filesrv.speag.com/outbox/'
+    var publicDir = 'https://outbox.zurichmedtech.com/maiz/';
+
+    var pathConverted = pathToConvert.replace(localDir, publicDir);
+    return pathConverted;
+  }
+
+  render() {
+    let pubPath = this.convertPathToPublic(this.props.showOutputDataReducer);
+    return (
+      <div className="ResultsViewer">
+        <Rnd
+          style = {{
+            color: this.props.color,
+            backgroundColor: this.props.backgroundColor,
+            borderStyle: 'solid',
+            opacity: .85
+          }}
+          visibility = {this.state.visible}
+          size = {{ width: this.state.width,  height: this.state.height }}
+          position = {{ x: this.state.x, y: this.state.y }}
+          onDragStop = {(e, d) => { this.setState({ x: d.x, y: d.y }) }}
+          onResize = {(e, direction, ref, delta, position) => {
+            this.setState({
+              width: ref.offsetWidth,
+              height: ref.offsetHeight,
+              ...position,
```

Comment:
a dangling comma ?

Reply:
I must admit it was a 'mistake', but according to the Airbnb JavaScript Style Guide, it is correct:
https://github.com/airbnb/javascript#commas

I should be consistent through the entire code though!

Reply:
well it just means you rely on a transpilation step ... from a coding point of view the dangling comma is certainly preferable. most frameworks transpile with babel anyway theses days, so I guess you are fine